pkg/async/notifications/implementations/sandbox_processor.go

```diff
@@ -0,0 +1,62 @@
+package implementations
+
+import (
+	"context"
+	"time"
+
+	"github.com/flyteorg/flyteadmin/pkg/async"
+	"github.com/flyteorg/flyteadmin/pkg/async/notifications/interfaces"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/golang/protobuf/proto"
+)
+
+type SandboxProcessor struct {
+	email   interfaces.Emailer
+	msgChan chan []byte
+}
+
+func (p *SandboxProcessor) StartProcessing() {
+	for {
+		logger.Warningf(context.Background(), "Starting SandBox notifications processor")
+		err := p.run()
+		logger.Errorf(context.Background(), "error with running processor err: [%v] ", err)
+		time.Sleep(async.RetryDelay)
+	}
+}
+
+func (p *SandboxProcessor) run() error {
+	var emailMessage admin.EmailMessage
+
+	for {
+		select {
+		case msg := <-p.msgChan:
+			err := proto.Unmarshal(msg, &emailMessage)
+			if err != nil {
+				logger.Errorf(context.Background(), "error with unmarshalling message [%v]", err)
+				return err
+			}
+
+			err = p.email.SendEmail(context.Background(), emailMessage)
+			if err != nil {
+				logger.Errorf(context.Background(), "error with sendemail message [%v] ", err)
```

Comment:
can we log the failed email message here too?

Reply:
Yes I will